core/src/main/java/org/apache/brooklyn/core/config/MapConfigKey.java

```diff
@@ -126,7 +126,7 @@ protected MapConfigKey(Builder<V> builder) {
         // Note: it's intentionally possible to have default values that are not valid
         // per the configured constraint. If validity were checked here any class that
         // contained a weirdly-defined config key would fail to initialise.
-        this.constraint = checkNotNull(builder.constraint, "constraint");
+        this.constraint = checkNotNull(builder.constraint, "constraints");
```

Comment:
Probably a leftover from reverting - the field name is still called `constraint`.